src/tests/modules/bson.rs

```diff
@@ -306,3 +307,18 @@ fn system_time() {
         0
     );
 }
+
+#[test]
+fn debug_print() {
+    #[derive(Debug)]
+    struct Example {
+        a: i32,
+        b: String,
+    }
+
+    let doc = doc! { "a": 1, "b": "data" };
```

Comment:
Can we include some BSON-specific types in here to ensure they get pretty-printed as well? (e.g. an ObjectID and a subdocument).

Also, I noticed while trying this out that we have an existing issue where such types have their type printed twice in the debug mode, e.g.:
```
{ a: 1, b: "data", c: { bool: true, date: Date("2021-06-07 20:20:26.905 UTC") } }

// gets debug printed as
Document({
    "a": Int32(
        1,
    ),
    "b": String(
        "data",
    ),
    "c": Document(
        Document({
            "bool": Boolean(
                true,
            ),
            "date": DateTime(
                DateTime(
                    2021-06-07T20:20:26.905Z,
                ),
            ),
        }),
    ),
})
```
Could we fix that up as part of this work too? I think it will require a custom `Debug` implementation for `Bson`, since the derived one is what's duplicating the inner type name.

Reply:
Ahh, interesting, thanks for the catch! I'll get to work on this when I'm done with my current ticket.